### PR TITLE
Support a closeTimeout prop for useTooltip

### PR DESCRIPTION
**Pull request by seri**
# Summary

Alex suggested that: 

> a closeTimeout that defaults to 0 but can be increased to leave the tooltip visible for a bit longer for these cases where there's something we want to click

This is very helpful when the tooltip contains a link.

**Main Story:** [ch67249](https://app.clubhouse.io/skyverge/story/67249/kodiak-tooltip-to-support-a-timeout-before-closing-the-portal)

## :vertical_traffic_light: Acceptance criteria

- [x] There is a story showcasing how to have the tooltip close 5 seconds after the mouse leaves

<a name="implementation-tasks"></a>

## Implementation Tasks

<a name="deployment"></a>

### Before merge to master

**Note**: _Code merged to master should be safe to automatically deploy to production as-is._

- [ ] **[QA]** User-testing/quality assurance done

### CODE REVIEW

**Comment by seri** [comment]
@alexanderchan @wesleycole i just ran `lerna publish` 😂 i think i'm supposed to wait for approval before that command

**Review by alexanderchan — CHANGES_REQUESTED** [review]
It's looking pretty good, just a weird edge case and a little update to get the click outside events triggering on click.

**Comment by alexanderchan** [comment]
> @alexanderchan @wesleycole i just ran `lerna publish` 😂 i think i'm supposed to wait for approval before that command

ahh no worries, we should probably try to automate the package deployment side of things to simplify this all maybe with gh actions.  

We might also consider moving from lerna to just yarn workspaces + changesets.  This is emotion's pr for the changeset side of things: https://github.com/emotion-js/emotion/pull/1343/files

**Review by alexanderchan — APPROVED** [review]
It looks good!  I had run build but forgot to reset to dev which is why I tried to push something with a change but it ended up being unnecessary🤦 

On the flip side, `yarn publish` should work to publish until we can automate.

File: packages/tooltip/src/useTooltip.ts
Comment (alexanderchan) [inline packages/tooltip/src/useTooltip.ts:68]: @seri this is really minor but I was just re-reading https://overreacted.io/making-setinterval-declarative-with-react-hooks/ and it reminded me of a small edge case to prevent getting an extra react dev error "Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in a useEffect cleanup function.".  This edge case happens if the popup is open, and we unmount the component (like hovering over something then changing tabs).

When the hook is dismounted we should also cleanup the timeout.  So to do this we'd need a 

```
React.useEffect(function cleanupTimeOut {
 return () => {
    clearTimeout(timeoutIdRef.current)
 }
})
```

**Pull request merged**
